src/basic/fd-util.h

```diff
@@ -126,6 +127,20 @@ static inline char *format_proc_fd_path(char buf[static PROC_FD_PATH_MAX], int f
 #define FORMAT_PROC_FD_PATH(fd) \
         format_proc_fd_path((char[PROC_FD_PATH_MAX]) {}, (fd))
 
+/* The maximum length a buffer for a /proc/self/fd/<fd> path needs */
+#define PROC_PID_FD_PATH_MAX \
+        (STRLEN("/proc//fd/") + DECIMAL_STR_MAX(pid_t) + DECIMAL_STR_MAX(int))
+
+static inline char *format_proc_pid_fd_path(char buf[static PROC_PID_FD_PATH_MAX], pid_t pid, int fd) {
```

Comment:
would we ever want to actually use a pid that is not our own here?

seems to me we should just change FORMAT_PROC_FD_PATH() to just always generate the pid based format (with an explanation why), that has no drawbacks i can see